src/github_repo_request_local.py

```diff
@@ -0,0 +1,74 @@
+import subprocess
+from pathlib import Path
+import pandas as pd
+from loguru import logger
+import shutil
+
+
+def count_and_list_test_files(directory):
+    logger.info(f"Processing the directory {directory}")
+    test_files = []
+    # Define a set of file extensions to exclude
+    excluded_extensions = {
+        ".txt",
+        ".md",
+        ".h",
+        ".xml",
+        ".html",
+        ".json",
+        ".png",
+        ".jpg",
+    }
+    for path in directory.rglob("*"):
+        # Check if the path is a file and either the file or its parent directory contains 'test'
+        if path.is_file() and (
+            "test" in path.name.lower() or "test" in str(path.parent).lower()
+        ):
+            # Exclude files with certain extensions
+            if path.suffix not in excluded_extensions:
+                test_files.append(str(path))
+    return len(test_files), test_files
+
+
+updated_csv_path = Path("../data/updated_local_github_df_test_count.csv")
+if updated_csv_path.exists():
+    logger.info("Resuming from previously saved progress.")
+    df = pd.read_csv(updated_csv_path)
+else:
+    csv_file_path = Path("../data/local_github_df_test_count.csv")
+    df = pd.read_csv(csv_file_path)
+    df = df.drop("Unnamed: 0", axis=1)
+    df["testfilecountlocal"] = -1  # Initialize if first run
+
+for index, row in df.iterrows():
+    if row["testfilecountlocal"] != -1:
+        continue  # Skip processed repositories
+
+    repo_url = row["repourl"]
+    repo_name = Path(repo_url.split("/")[-1]).stem
+    clone_dir = Path.home() / "data" / "cloned_repos" / repo_name
```

Comment:
This hardcoded path is good enough to get us started; in future it might be worth making this configurable e.g. so the cloned repos could be created on a separate volume with lots of available space rather than in the home folder's drive.

Reply:
When required, Python's `argparse` module can be used or get the clone directory from an environment variable, with a default fallback.

Reply:
by environment variable do you mean one that's set using `export ENV_VAR=/home/julian/...` and/or one that's set using a command-line option such as `--working-dir=/home/julian/...`? I agree that either could be used especially if argparse supports both. If we only support one of these I'd prefer us to use command-line options for the working directory/folder as it's easier to see the setting when the program is run.

Reply:
FYI I'm for using environment variables for secrets such as passwords rather than needing these to be passed as command line arguments where they're easier to be inadvertently logged.

Reply:
From my understanding both can be implemented . Can we leave environment variables for secrets as you mentioned? Because I also have used them in the same manner. I will resolve this by using the command-line option 

- Enabled configuration of the clone directory path through the --clone-dir command-line argument. This adjustment makes the script more flexible, accommodating environments where the default clone location is not suitable or preferred.
- Code has been pushed to the branch